### ISSUE TITLE
[Bug]: HTML In App back navigation not working correctly

### ISSUE THREAD

**Issue by pedromfmachado** [braze-inc/braze-android-sdk#39]
### Braze Android SDK Version

30.4.0

### Steps To Reproduce

1. Enable the predictive back behaviour by adding
    - Open AndroidManifest.xml
    - Add the `android:enableOnBackInvokedCallback="true"` property under the `<Application>` tag
2. Open an HTML in app message using
3. Navigate back using the device back button (physical or otherwise)

### Expected Behavior

- The html view closes leading the user to the previous screen

### Actual Incorrect Behavior

- The back behaviour is not intercepted, leading to the closure of the html view and the previous one

### Additional Information

This might be due to the fact that [`InAppMessageHtmlBaseView.kt`](https://github.com/braze-inc/braze-android-sdk/blob/master/android-sdk-ui/src/main/java/com/braze/ui/inappmessage/views/InAppMessageHtmlBaseView.kt) is still using `KEYCODE_BACK` to intercept back navigation (see [InAppMessageHtmlBaseView.kt#L274](https://github.com/braze-inc/braze-android-sdk/blob/3bae4d276868cbe13770db0e4df70d8930608f02/android-sdk-ui/src/main/java/com/braze/ui/inappmessage/views/InAppMessageHtmlBaseView.kt#L274)). This specific behaviour is not supported as described on the [Add support for the predictive back gesture](https://developer.android.com/guide/navigation/custom-back/predictive-back-gesture) guide:
> Note: KeyEvent.KEYCODE_BACK is not deprecated as there are some supported use cases of KeyEvent.KEYCODE_BACK; however, intercepting back events from KeyEvent.KEYCODE_BACK is no longer supported.

**Comment by bryanlogan** [comment]
Thanks for the bug report. I've opened a ticket internally to look at it. 
As a workaround until we have the final fix, you could put this:

```
BrazeInAppMessageManager.getInstance().apply { 
      if (isCurrentlyDisplayingInAppMessage) {
           hideCurrentlyDisplayingInAppMessage(true)
      }
}
```

As part of your `OnBackPressedDispatcher`

**Comment by shaubert** [comment]
Hi @bryanlogan, do you have any ETA for the fix?

**Comment by bryanlogan** [comment]
This was fixed in 32.1.0.
